From 9859e53e49f9b934f649289dada9109e6a7b4399 Mon Sep 17 00:00:00 2001
From: Aaron Fabbri <fabbri@cloudera.com>
Date: Wed, 25 May 2016 21:01:57 -0700
Subject: [PATCH 1595/2748] IMPALA-2607 CLOUDERA-BUILD S3: need to determine
 the right default for fs.s3a.connection.maximum

Increase fs.s3a.connection.maximum config default from 15 to 100 TCP
connections.  This allows applications like Impala to keep more files (input
streams) open at a time.

Based on discussion with Impala devs, there isn't a real upper bound on number
of files open per-impala process.   It sounds like there is no real upper bound
on the number of open files (and thus connections from s3a's underlying Apache
httpclient connection pool).

Change-Id: If7a588dd9e1ba96de2b5e1def701fde3554f8c4b
---
 .../src/main/resources/core-default.xml            |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/resources/core-default.xml b/hadoop-common-project/hadoop-common/src/main/resources/core-default.xml
index 8a4fc0b..91930f9 100644
--- a/hadoop-common-project/hadoop-common/src/main/resources/core-default.xml
+++ b/hadoop-common-project/hadoop-common/src/main/resources/core-default.xml
@@ -753,7 +753,7 @@ for ldap providers in the same way as above does.
 
 <property>
   <name>fs.s3a.connection.maximum</name>
-  <value>15</value>
+  <value>100</value>
   <description>Controls the maximum number of simultaneous connections to S3.</description>
 </property>
 
-- 
1.7.9.5

